pkg/machinery/config/types/runtime/fs_scrub.go

```diff
@@ -0,0 +1,157 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package runtime
+
+//docgen:jsonschema
+
+import (
+	"fmt"
+	"net/url"
+	"time"
+
+	"github.com/siderolabs/talos/pkg/machinery/config/config"
+	"github.com/siderolabs/talos/pkg/machinery/config/internal/registry"
+	"github.com/siderolabs/talos/pkg/machinery/config/types/meta"
+	"github.com/siderolabs/talos/pkg/machinery/config/validation"
+)
+
+// FilesystemScrubKind is a watchdog timer config document kind.
+const FilesystemScrubKind = "FilesystemScrubConfig"
+
+func init() {
+	fmt.Println("TEST! FilesystemScrubKind: ", FilesystemScrubKind)
+	registry.Register(FilesystemScrubKind, func(version string) config.Document {
+		switch version {
+		case "v1alpha1":
+			return &FilesystemScrubV1Alpha1{}
+		default:
+			return nil
+		}
+	})
+}
+
+// Check interfaces.
+var (
+	_ config.RuntimeConfig = &FilesystemScrubV1Alpha1{}
+	_ config.NamedDocument = &FilesystemScrubV1Alpha1{}
+	_ config.Validator     = &FilesystemScrubV1Alpha1{}
+)
+
+// Timeout constants.
+const (
+	MinScrubPeriod     = 10 * time.Second
+	DefaultScrubPeriod = 24 * 7 * time.Hour
+)
+
+// FilesystemScrubV1Alpha1 is a filesystem scrubbing config document.
```

Comment:
Docs must mention this feature is experimental and perhaps link to kernel docs about this